modules/library/main/src/main/java/org/geotools/data/ows/ControlledHttpClientFactory.java

```diff
@@ -0,0 +1,25 @@
+/*
+ *    GeoTools - The Open Source Java GIS Toolkit
+ *    http://geotools.org
+ *
+ *    (C) 2020, Open Source Geospatial Foundation (OSGeo)
+ *
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License as published by the Free Software Foundation;
+ *    version 2.1 of the License.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ */
+package org.geotools.data.ows;
+
+/** @author ImranR */
+public class ControlledHttpClientFactory {
+
+    public static synchronized HTTPClient wrap(HTTPClient client) {

```

Comment:
understood will add. synchronized is used to handle concurrency

Reply:
This will kill the whole system concurrency. What data structure do you have to protect from concurrency? Find it and secure that one only, this method does not need to be synchronized.

Reply:
makes sense, will remove synchronized